geode-core/src/main/java/org/apache/geode/cache/query/internal/LinkedStructSet.java

```diff
@@ -63,8 +62,13 @@ public LinkedStructSet(StructTypeImpl structType) {
 
   @Override
   public boolean equals(Object other) {
-    return other instanceof SortedStructSet
-        && this.structType.equals(((SortedStructSet) other).structType) && super.equals(other);
+    if (!(other instanceof SortedStructSet)) {
+      return false;
+    }
+    if (!this.structType.equals(((SortedStructSet) other).structType)) {
+      return false;
+    }
+    return super.equals(other);
```

Comment:
So, by merging develop into your branch and resolving the conflicts, some of your original changes got reverted.